Assets/Scripts/Level/TutorialTriggerGroup.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Assets.Scripts.Level
+{
+    public class TutorialTriggerGroup : MonoBehaviour
+    {
+
+        public TutorialTrigger[] triggers;
+        public InGameUI gameUI;
+        public PlayerProgress playerProgress;
+
+        // Use this for initialization
+        void Start()
+        {
+            playerProgress = GameObject.FindWithTag("Player").GetComponent<PlayerProgress>();

```

Comment:
nice find, this is the most efficient of the "FindWithXXX" functions. Fun fact, you don't need "GameObject." here, as there's syntactic sugar added.

Reply:
Yeah, I saw that it was much more efficient than generic Find. I'll remove the GameObject call. One thing to note is that I tagged the player character. It wasn't tagged beforehand. But I figure as we add more scripts the player character is likely to be one of the most referenced objects, and we had the tag already present.

Reply:
I'm perfectly okay with this solution. my only complaint is the use of a magic string. Move it to the PlayerConstants and i'll be happy :D

Reply:
Actually, it's the other way around. GameObject is needed for FindWithTag. It can be removed for FindObjectOfType. I don't know why that's the case, but it appears to be the case.